zipline/assets/assets.py

```diff
@@ -809,6 +882,79 @@ def lookup_future_symbol(self, symbol):
             raise SymbolNotFound(symbol=symbol)
         return self.retrieve_asset(data['sid'])
 
+    def lookup_by_supplementary_mapping(self, mapping_type, value, as_of_date):
+        try:
+            owners = self.supplementary_map[
```

Comment:
Yea, I was also considering whether to load both directions (sid->identifier, identifier->sid) together when one direction is first requested, re-using all the query results, instead of reading the whole table again. I ended up thinking leave it as is, as I don't think it's a huge data set memory-wise, and it then becomes just a question of when to load it, which I don't have a strong opinion about. Curious if you have an opinion.